### PR TITLE
[go_router_builder] Fix #106121 and #116837 - Fix compile error on the generated code when using $extra and non nullable enums queryParams

### PR DESCRIPTION
**Pull request by dancamdev**
Issues Fixed:
[https://github.com/flutter/flutter/issues/116837](https://github.com/flutter/flutter/issues/116837)

Relevant
[https://github.com/flutter/flutter/issues/106121](https://github.com/flutter/flutter/issues/106121)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
I think this only fixes https://github.com/flutter/flutter/issues/106121 partially. The main issue is the extra may be overloaded with GoRouteData, but that can be a separate pr.

LGTM

**Comment by dancamdev** [comment]
I also think there's room for improvement on how extras are handled.

I'll work on a new PR with proposed changes, I've got a couple of ideas to make improve DX.

File: packages/go_router/lib/src/misc/extensions.dart [hunk 1]
```diff
@@ -89,4 +89,7 @@ extension GoRouterHelper on BuildContext {
         queryParams: queryParams,
         extra: extra,
       );
+
+  /// Returns true if the route is registered within the [GoRouter].
+  bool hasRoute(String name) => GoRouter.of(this).hasRoute(name);
```
Comment (chunhtai) [inline packages/go_router/lib/src/misc/extensions.dart:94]: It seems like this is not relevant to this PR?

**Review by chunhtai — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@dancamdev Are you planning on updating this PR based on the feedback above?

**Comment by dancamdev** [comment]
Sure, hopefully I find time this week

**Comment by 5hirish** [comment]
Meanwhile is there any workaround we can use here?

**Comment by chunhtai** [comment]
@dancamdev are you still planning on coming back to this pr?

**Review by dancamdev — COMMENTED** [review]
Ok, I somehow messed up with an unrelated commit - I removed all code having nothing to do with this specific issue.

**Comment by chunhtai** [comment]
The ci is failing, can you fix the failing test?

**Comment by chunhtai** [comment]
(triage) Hi @dancamdev are you still plan to work on this PR? If not we can close this PR for now and reopen when you have time to work on it.

**Comment by dancamdev** [comment]
It looks like tests are timing out on my end.

Getting this error on Windows

```
hell: [ERROR:flutter/shell/common/shell.cc(89)] Dart Error: error: import of dart:mirrors is not supported in the current Dart runtime
Shell: [ERROR:flutter/runtime/dart_isolate.cc(144)] Could not prepare isolate.
```

I'll try to run it on Linux and see if that works

**Comment by stuartmorgan** [comment]
@dancamdev Are you still planning on updating this to address the test issue?

**Comment by chunhtai** [comment]
Hi @dancamdev I will close this pr for now. Feel free to reopen once you have time to work on this

**Pull request closed without merging**